### ISSUE TITLE
"dev" branch : one mistake in classification in logical_operations

### ISSUE THREAD

**Issue by shleym2000** [Artelnics/opennn#130]
The **logical_operations** example now produces results, which have one "mistake" in it.

The expected results are what is in input data:
```
X;Y;AND;OR;NAND;NOR;XNOR;XNOR
1;1;1;1;0;0;0;1
1;0;0;1;1;0;1;0
0;1;0;1;1;0;1;0
0;0;0;0;1;1;0;1
```
Now the result produced by this example achieved after 246 training epochs:

> Epoch 245;
> Parameters norm: 7.63786
> Training error: 0.696064
> Gradient norm: 1.90625e-08
> Learning rate: 1.25408
> Elapsed time: 00:00:01
> propagation
> Combinations
> Activations
> Epoch 246: Minimum parameters increment norm reached.
> Parameters increment norm: 0
> Parameters norm: 7.63786
> Training error: 0.696064
> Gradient norm: 1.2807e-09
> Learning rate: 0
> Elapsed time: 00:00:01
> X Y AND OR NAND NOR XOR XNOR
> X = 1 Y = 1
> 1 1    0.480029      0.4951  0.00542991 7.31613e-05   0.0192939 **7.48495e-05**
> X = 1 Y = 0
> 1 0 0.00847967   0.336458    0.33079 0.00442591   0.315375 0.00447116
> X = 0 Y = 1
> 0 1 0.00847967   0.336458    0.33079 0.00442591   0.315375 0.00447116
> X = 0 Y = 0
> 0 0 5.33778e-05  0.00262698    0.336453    0.323794   0.0134415    0.323631
> 

If we consider results greater than 0.3 as 1 and results lower than 0.01, **results meet expectations in all cases _except one_** in upper right corner of the results:
1 XNOR 1 **should be 1**, but the produced result is 7.48495e-05, which is by classification above is 0.

**Comment by shleym2000** [comment]
This test has 2 input nodes and 6 output nodes.
The result above with 1 out of 24 outputs determined incorrectly was produced when the number of hidden neurons was 6.
I tried to experiment with different _hidden_neurons_number_ values.

hidden_neurons_number = 5 gave 2 incorrect results, which might be explained as lack of capacity

> Epoch 149: Minimum parameters increment norm reached.
> Parameters increment norm: 0
> Parameters norm: 6.33718
> Training error: 0.756398
> Gradient norm: 2.31576e-09
> Learning rate: 0
> Elapsed time: 00:00:00
> Jul  5 2020 07:52:25 : **hidden_neurons_number = 5**
> X Y AND OR NAND NOR XOR XNOR
> X = 1 Y = 1
> 1 1 **0.000936557**    0.905737   0.0245291  0.00013042   0.0685323 **0.000135083**
> X = 1 Y = 0
> 1 0 0.00381295   0.350397   0.328089 0.00474142   0.308124 0.00483537
> X = 0 Y = 1
> 0 1 0.00381295   0.350397   0.328089 0.00474142   0.308124 0.00483537
> X = 0 Y = 0
> 0 0 0.00492023 0.00288273   0.334887   0.321657  0.0141802   0.321473
> bye

hidden_neurons_number = 7- same 1 incorrect reply as with 6 hidden neurons

> Epoch 316: Minimum parameters increment norm reached.
> Parameters increment norm: 0
> Parameters norm: 7.58922
> Training error: 0.695953
> Gradient norm: 1.81722e-09
> Learning rate: 0
> Elapsed time: 00:00:01
> Jul  5 2020 07:59:01 : **hidden_neurons_number = 7**
> X Y AND OR NAND NOR XOR XNOR
> X = 1 Y = 1
> 1 1    0.480618    0.494871  0.00497879 5.51516e-05   0.0194208 **5.60847e-05**
> X = 1 Y = 0
> 1 0 0.00845634   0.336754   0.331046 0.00446723   0.314774 0.00450218
> X = 0 Y = 1
> 0 1 0.00845634   0.336754   0.331046 0.00446723   0.314774 0.00450218
> X = 0 Y = 0
> 0 0 3.28976e-05  0.00220799    0.336183    0.324174    0.013353     0.32405
> bye
> 

hidden_neurons_number = 8- same 1 incorrect reply as with 6 hidden neurons

> Epoch 206: Minimum parameters increment norm reached.
> Parameters increment norm: 0
> Parameters norm: 6.18268
> Training error: 0.755809
> Gradient norm: 4.75505e-09
> Learning rate: 0
> Elapsed time: 00:00:01
> Jul  5 2020 08:04:02 : **hidden_neurons_number = 8**
> X Y AND OR NAND NOR XOR XNOR
> X = 1 Y = 1
> 1 1 0.000380133    0.912055   0.0182459 4.51698e-05   0.0692281 **4.58093e-05**
> X = 1 Y = 0
> 1 0 0.00316307   0.351521   0.329763 0.00441045   0.306696 0.00444598
> X = 0 Y = 1
> 0 1 0.00316307   0.351521   0.329763 0.00441045   0.306696 0.00444598
> X = 0 Y = 0
> 0 0  0.0045326 0.00201528   0.334487   0.322247   0.014543   0.322175
> bye

**Comment by shleym2000** [comment]
Results from the current code as of August 7, 2020. Wrong numbers are in **bold**. Both are in the first row and they should be greater than 0.3 threshold.

> Iteration 70: Gradient norm goal reached.
> Parameters norm: 6.28542
> Training error: 0.756226
> Gradient norm: 0.000893716
> Learning rate: 3.25296
> Elapsed time: 00:00:00
> X Y AND OR NAND NOR XOR XNOR
> X = 1 Y = 1
> 1 1 **0.000703005**    0.912245   0.0183483 6.58319e-05   0.0685629 **7.51348e-05**
> X = 1 Y = 0
> 1 0 0.00410951   0.350344   0.327848 0.00452635   0.308235 0.00493638
> X = 0 Y = 1
> 0 1 0.00397886    0.35113   0.328775 0.00439575   0.306926 0.00479455
> X = 0 Y = 0
> 0 0 0.00534074 0.00226776   0.332625   0.323735  0.0139914   0.322039
> bye

**Comment by jandres-artelnics** [comment]
Hi @shleym2000 ,

You're having a training error of 0.75 and that's pretty bad. Sounds like there's something really wrong.

I took a look at the example and I found that the neural network used was a classification project.

Try using an approximation project instead and it'll go fine!

This is a pic of the results (using dev)

![image](https://user-images.githubusercontent.com/59603057/92105260-dfb82300-ede2-11ea-990f-88c96fe39320.png)

Thank you for your comment!

I'll close de issue. Please feel free to open a new issue is anything else is not working.
